tests/persistence/test_persistence_leverage.py

```diff
@@ -0,0 +1,638 @@
+        Short trade
+        fee: 0.25% base
+        interest_rate: 0.05% per 4 hrs
+        open_rate: 0.00004099 base
+        close_rate: 0.00004173 base
+        amount: 91.99181073 * leverage(3) = 275.97543219 crypto
+        stake_amount: 0.0037707443218227
+        borrowed: 0.0075414886436454 base
+        time-periods: 10 minutes = 2
+        interest: borrowed * interest_rate * time-periods
+            = 0.0075414886436454 * 0.0005 * 2 = 7.5414886436454e-06 crypto
+        open_value: (amount * open_rate) + (amount * open_rate * fee)
+            = (275.97543219 * 0.00004099) + (275.97543219 * 0.00004099 * 0.0025)
+            = 0.01134051354788177
+        close_value: (amount_closed * close_rate) - (amount_closed * close_rate * fee) - interest
+        = (275.97543219 * 0.00001234) - (275.97543219 * 0.00001234 * 0.0025) - 7.5414886436454e-06
+            = 0.0033894815024978933
+        = (275.97543219 * 0.00001234) - (275.97543219 * 0.00001234 * 0.003) -  7.5414886436454e-06
+            = 0.003387778734081281
+        = (275.97543219 * 0.00004173) - (275.97543219 * 0.00004173 * 0.005) -  7.5414886436454e-06
+            = 0.011451331022718612
+    """
+    trade = Trade(
+        pair='ETH/BTC',
+        stake_amount=0.0037707443218227,
+        amount=5,
+        open_rate=0.00004099,
+        fee_open=fee.return_value,
+        fee_close=fee.return_value,
+        open_date=datetime.utcnow() - timedelta(hours=0, minutes=10),
+        interest_rate=0.0005,
+        leverage=3.0,
+        exchange='kraken',
+        interest_mode=InterestMode.HOURSPER4
+    )
+    trade.open_order_id = 'close_trade'
+    trade.update(market_lev_buy_order)  # Buy @ 0.00001099
+    # Get the close rate price with a custom close rate and a regular fee rate
+    assert isclose(trade.calc_close_trade_value(rate=0.00001234), 0.0033894815024978933)
+    # Get the close rate price with a custom close rate and a custom fee rate
+    assert isclose(trade.calc_close_trade_value(rate=0.00001234, fee=0.003), 0.003387778734081281)
+    # Test when we apply a Sell order, and ask price with a custom fee rate
+    trade.update(market_lev_sell_order)
+    assert isclose(trade.calc_close_trade_value(fee=0.005), 0.011451331022718612)
+
+
+@pytest.mark.usefixtures("init_persistence")
+def test_update_with_binance_lev(limit_lev_buy_order, limit_lev_sell_order, fee, caplog):
+    """
+        10 minute leveraged limit trade on binance at 3x leverage
+
+        Leveraged trade
+        fee: 0.25% base
+        interest_rate: 0.05% per day
+        open_rate: 0.00001099 base
+        close_rate: 0.00001173 base
+        amount: 272.97543219 crypto
+        stake_amount: 0.0009999999999226999 base
+        borrowed: 0.0019999999998453998  base
+        time-periods: 10 minutes(rounds up to 1/24 time-period of 1 day)
+        interest: borrowed * interest_rate * time-periods
+                    = 0.0019999999998453998 * 0.0005 * 1/24 = 4.166666666344583e-08 base
```

Comment:
The amount of decimals used in these tests (all tests, not just this one) makes it near impossible to sanely (without spending days doing the math) determine if the calculation is correct or not.

Reply:
I tried my best to fix it, still some values with a lot of decimals though